pkg/trace/traceutil/normalize.go

```diff
@@ -232,6 +238,126 @@ func NormalizeTag(v string) string {
 	return string(tag)
 }
 
+// NormalizeValue applies some normalization to ensure the tags match the backend requirements.
+func NormalizeValue(v string) string {
+	// Fast path: Check if the tag is valid and only contains ASCII characters,
+	// if yes return it as-is right away. For most use-cases this reduces CPU usage.
+	// Pass true to allow digits as the first char, since we're normalizing the _value_ of a tag, not the whole tag key:value string.
+	if isNormalizedASCIITag(v, true) {
+		return v
+	}
+	// the algorithm works by creating a set of cuts marking start and end offsets in v
+	// that have to be replaced with underscore (_)
+	if len(v) == 0 {
+		return ""
+	}
+	var (
+		trim  int      // start character (if trimming)
+		cuts  [][2]int // sections to discard: (start, end) pairs
+		chars int      // number of characters processed
+	)
+	var (
+		i    int  // current byte
+		r    rune // current rune
+		jump int  // tracks how many bytes the for range advances on its next iteration
+	)
+	tag := []byte(v)
+	for i, r = range v {
+		jump = utf8.RuneLen(r) // next i will be i+jump
+		if r == utf8.RuneError {
+			// On invalid UTF-8, the for range advances only 1 byte (see: https://golang.org/ref/spec#For_range (point 2)).
+			// However, utf8.RuneError is equivalent to unicode.ReplacementChar so we should rely on utf8.DecodeRune to tell
+			// us whether this is an actual error or just a unicode.ReplacementChar that was present in the string.
+			_, width := utf8.DecodeRune(tag[i:])
+			jump = width
+		}
+		// fast path; all letters (and colons) are ok
+		switch {
+		case r >= 'a' && r <= 'z' || r >= '0' && r <= '9' || r == ':':
```

Comment:
Include digits in this case so that it's OK to have them as a start char for the value.